yazi-plugin/preset/plugins/extract.lua

```diff
@@ -47,23 +57,45 @@ function M:try_with(url, pwd)
 		fail("Failed to determine a temporary directory for %s", url)
 	end
 
+	os.execute("mkdir -p " .. tostring(tmp))
 	local archive = require("archive")
-	local child, code = archive:spawn_7z { "x", "-aou", "-p" .. pwd, "-o" .. tostring(tmp), tostring(url) }
-	if not child then
-		fail("Spawn `7z` and `7zz` both commands failed, error code %s", code)
-	end
+	if self:is_tar_compatible(tostring(url)) then
+		local child, code = archive:spawn_tar { "-xvf", tostring(url), "-C", tostring(tmp) }
```

Comment:
```suggestion
		local child, code = archive:spawn_tar { "-xf", tostring(url), "-C", tostring(tmp) }
```

Any reason for specifying the `-v` switch?

Reply:
Ur right, it's an optional one for printing the output of filename, gonna delete this.